taker-frontend/src/components/ConfirmOrderModal.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+    Button,
+    HStack,
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    ModalFooter,
+    ModalHeader,
+    ModalOverlay,
+    Table,
+    TableCaption,
+    Tbody,
+    Td,
+    Text,
+    Tooltip,
+    Tr,
+} from "@chakra-ui/react";
+import * as React from "react";
+import { CfdOrderRequestPayload } from "../types";
+import BitcoinAmount from "./BitcoinAmount";
+import DollarAmount from "./DollarAmount";
+
+interface Props {
+    orderId: string;
+    isOpen: boolean;
+    onClose: any;
+    onSubmit: (req: CfdOrderRequestPayload) => void;
+    isSubmitting: boolean;
+    setQuantity: (req: any) => void;
+    setUserHasEdited: (value: any) => void;
+    confirmRef: any;
+    quantity: number;
+    minQuantity: number;
+    askPriceAsNumber?: number;
+    margin: number;
+    leverage: number;
+    liquidationPriceAsNumber: number | undefined;
+    feeForFirstSettlementInterval: number;
+    fundingRateHourly: string;
+    fundingRateAnnualized: string;
```

Comment:
This seems a bit excessive, do we really need all of these?

`setQuantity` for example is only used to set the input field back to `minQuantity`. This component doesn't even know about the input field so I'd set the call to `setQuantity` should happen in the callback that the parent passes in for `onClose`.

Reply:
We should do a similar analysis for all other props.